src/macports1.0/macports.tcl

```diff
@@ -1070,9 +1073,10 @@ proc mportinit {{up_ui_options {}} {up_options {}} {up_variations {}}} {
     set macosx_version $macos_version_major
 
     # Check that the current platform is the one we were configured for, otherwise need to do migration
-    if {($os_platform ne $macports::autoconf::os_platform) || ($os_platform eq "darwin" && $os_major != $macports::autoconf::os_major)} {
+    set skip_migration_check [expr {[info exists macports::global_options(ports_no_migration_check)] && $macports::global_options(ports_no_migration_check)}]
+    if {!$skip_migration_check && [migrate::needs_migration]} {
         ui_error "Current platform \"$os_platform $os_major\" does not match expected platform \"$macports::autoconf::os_platform $macports::autoconf::os_major\""
-        ui_error "If you upgraded your OS, please follow the migration instructions: https://trac.macports.org/wiki/Migration"
+        ui_error "Please run 'sudo port migrate' or follow the migration instructions: https://trac.macports.org/wiki/Migration"
```

Comment:
This is probably the line we need to revert if we don't consider this ready for prime-time before we cut the next release.